crates/tx-graph2/src/connectors/claim_contest_connector.rs

```diff
@@ -0,0 +1,439 @@
+//! This module contains the connector between `Claim`, `UncontestedPayout`, and `Contest`.
+
+use bitcoin::{
+    opcodes,
+    psbt::Input,
+    relative,
+    taproot::{LeafVersion, TaprootSpendInfo},
+    Network, ScriptBuf, TxOut,
+};
+use secp256k1::{schnorr, XOnlyPublicKey};
+use strata_bridge_primitives::scripts::prelude::{create_taproot_addr, finalize_input, SpendPath};
+
+/// Connector output between `Claim` and:
+/// 1. `UncontestedPayout`, and
+/// 2. `Contest`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ClaimContestConnector {
+    network: Network,
+    n_of_n_pubkey: XOnlyPublicKey,
+    watchtower_pubkeys: Vec<XOnlyPublicKey>,
+    delta_contest: relative::LockTime,
+}
+
+impl ClaimContestConnector {
+    /// Creates a new connector.
+    pub const fn new(
+        network: Network,
+        n_of_n_pubkey: XOnlyPublicKey,
+        watchtower_pubkeys: Vec<XOnlyPublicKey>,
+        delta_contest: relative::LockTime,
+    ) -> Self {
+        Self {
+            network,
+            n_of_n_pubkey,
+            watchtower_pubkeys,
+            delta_contest,
+        }
+    }
+
+    /// Returns the number of watchtowers for the connector.
+    pub const fn n_watchtowers(&self) -> usize {
+        self.watchtower_pubkeys.len()
+    }
+
+    /// Returns the delta contest relative timelock of the connector.
+    ///
+    /// The sequence of the input and the global locktime must be
+    /// large enough to cover this value.
+    pub const fn delta_contest(&self) -> relative::LockTime {
```

Comment:
how about `contest_timelock`?

Reply:
So we write `timelock` instead of `delta`? Is this a general rule?